homeassistant/components/tuya/sensor.py

```diff
@@ -739,6 +739,74 @@ class TuyaSensorEntityDescription(SensorEntityDescription):
             icon="mdi:progress-clock",
         ),
     ),
+    # Air Purifier
+    # https://developer.tuya.com/en/docs/iot/s?id=K9gf48r41mn81
+    "kj": (
+        TuyaSensorEntityDescription(
+            key=DPCode.FILTER,
+            name="Filter Utilization",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            icon="mdi:ticket-percent-outline",
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.FILTER_DAYS,
+            name="Remaining days of the filter",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            icon="mdi:ticket-confirmation-outline",
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.PM25,
+            name="Particulate Matter 2.5 µm",
+            device_class=SensorDeviceClass.PM25,
+            state_class=SensorStateClass.MEASUREMENT,
+            icon="mdi:molecule",
```

Comment:
An icon has been defined for PM25 because the unit of measurement in the documentation is null, this causes the device_class defined here to be replaced by none.

Reply:
Still, we should not set it :)
If the unit isn't set, we should contact tuya or the manufacturer for that device.

Reply:
Second thought.

What if we DO set it...

in the sensor code, there is a check to see if the unit is valid. If it is not valid, it removes the device & state class.
What about adding the reverse: If it IS valid, we remove the icon and leave the device & state class.

That way, we can satisfy both paths, without breaking general architectural Home Assistant rule?

Reply:
I believe you are referencing the line 844.
https://github.com/home-assistant/core/blob/5da923c34104c9bc8e894f136f07d53ce27d8580/homeassistant/components/tuya/sensor.py#L844
````
            # Unknown unit of measurement, device class should not be used.
            if self._uom is None:
                self._attr_device_class = None
                return
````

But it also has the line that handles this, as it would be handled in the line 832?
https://github.com/home-assistant/core/blob/5da923c34104c9bc8e894f136f07d53ce27d8580/homeassistant/components/tuya/sensor.py#L832

````
            if (
                self.native_unit_of_measurement is None
                or self.device_class not in DEVICE_CLASS_UNITS
            ):
                self._attr_device_class = None
                return
````

I didn't find where to manipulate the `StateClass` in Tuya's sensor.

Reply:
change:
```py
            # Unknown unit of measurement, device class should not be used.
            if self._uom is None:
                self._attr_device_class = None
                return
```

to

```py
            # Unknown unit of measurement, device class should not be used.
            if self._uom is None:
                self._attr_device_class = None
                return
			
			# If we still have a device class, we should not use an icon
            if self.device_class:
                self._attr_icon = None
```

Reply:
Changed. The icon for the `DeviceClass` was defined even with the unit of measure null.